pkg/collector/check/logreceiver.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package check
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/comp/logs/integrations/def"
+)
+
+var lr integrations.Component
```

Comment:
We spent time during last quarters to remove these kind of global components.

As these functions are not used yet, can you remove them and inject integrations.Component where it is needed?

Reply:
The global state is being used for this change: https://github.com/DataDog/datadog-agent/pull/26978. This was the best we way were were able to get the new component into python checks.

Reply:
Are there any other way for Python to access a Go object? Asking out of curiosity

Reply:
Not that I know of

Reply:
> Are there any other way for Python to access a Go object? Asking out of curiosity

Currently no. All of the checks code is stateless and access to agent state is done through global wrappers ([one of many examples](https://github.com/DataDog/datadog-agent/blob/579d60cbfc8635559338b19fa855bf59b6d10f00/pkg/collector/python/check_context.go#L28) and [it's usage](https://github.com/DataDog/datadog-agent/blob/579d60cbfc8635559338b19fa855bf59b6d10f00/pkg/collector/python/aggregator.go#L34))

I think there is a much wider refactoring effort that needs to be done here in order to make the collector (specifically the python bits) compatible with components.

Reply:
I added the [checkContext](https://github.com/DataDog/datadog-agent/blob/579d60cbfc8635559338b19fa855bf59b6d10f00/pkg/collector/python/check_context.go#L28) to allow passing components to functions like [SubmitMetric](https://github.com/DataDog/datadog-agent/blob/579d60cbfc8635559338b19fa855bf59b6d10f00/pkg/collector/python/aggregator.go#L34):
- `checkContext` is initialized in `NewPythonCheckLoader` so fields are always set when calling it from a python check.
- Having one global to update is easier than having several globals to update if we need to change this code in the future.

Would it be possible to use `checkContext` for you case? If not, I will be curious to know why so I can try to improve checkContext.

Reply:
Yep I think that should be possible, let me try and work it through.

Reply:
@ogaca-dd Made the change to move the logReceiver into checkContext from it's own file 👍